### PR TITLE
minor refactors around consensus and use actual is_short_range API

### PR DESCRIPTION
**Pull request by creativcoder**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

File: consensus/src/checkpoint.rs [hunk 1]
```diff
@@ -3,15 +3,12 @@
...
 use mina_crypto::base58::Base58Encodable;
 use mina_crypto::hash::{EpochSeed, StateHash};
 use mina_rs_base::protocol_state::ProtocolState;
 
-#[derive(Debug)]
-pub enum ConsensusErrTyp {
-    ConsensusInitFail,
-}
-pub fn init_checkpoints(genesis: &mut ProtocolState) -> Result<(), ConsensusErrTyp> {
+pub fn init_checkpoints(genesis: &mut ProtocolState) -> Result<(), ConsensusError> {
```
Comment (hanabi1224) [inline consensus/src/checkpoint.rs:11]: How about creating a Checkpoint trait and making this a trait method? In general should we prefer OO way in our api design or non-OO way?

File: consensus/src/common.rs [hunk 2]
```diff
@@ -117,11 +118,6 @@ pub trait Consensus {
     ) -> Result<&'a ProtocolStateChain, ConsensusError>;
 }
 
-// TODO: replace from checkpoint.rs
-fn is_short_range(_cs: &ProtocolStateChain) -> bool {
```
Comment (polarbeargo) [inline consensus/src/common.rs:121]: It's conflict with spec https://github.com/MinaProtocol/mina/blob/32f529d8d8d712a44ee75be66061ce08cbdc8924/docs/specs/consensus/README.md#532-short-range-fork-check

**Review by polarbeargo — CHANGES_REQUESTED** [review]

File: consensus/src/checkpoint.rs (hunk 1 above)
Comment (creativcoder) [reply consensus/src/checkpoint.rs:11]: Not sure how we could possibly enforce OO around every code, but definitely trait based APIs wherever possible (the rusty way). Yes am thinking of refactoring the checkpoint related parts, as part of this PR.

File: consensus/src/common.rs (hunk 2 above)
Comment (creativcoder) [reply consensus/src/common.rs:121]: yes, this has been replaced with the is_short_range function from `checkpoint.rs`

File: consensus/src/checkpoint.rs (hunk 1 above)
Comment (creativcoder) [reply consensus/src/checkpoint.rs]: I notice the `init_checkpoint` function is only being used in test cases at the moment, so i have moved it to under `tests/checkpoint.rs` within `mod tests {}`. There isn't really much to refactor in checkpoint.rs related to code design at the moment as we simply have a `is_short_range` function there. Based on how it's being used in `select_secure_chain` API, it feels having `is_short_range` as a free helper function is appropriate.

**Review by hanabi1224 — APPROVED** [review]

**Review by noot — APPROVED** [review]

File: consensus/tests/checkpoint.rs [hunk 3]
```diff
@@ -3,11 +3,44 @@
...
+            .staking_epoch_data
+            .lock_checkpoint = StateHash::default();
+        genesis
+            .body
+            .consensus_state
+            .staking_epoch_data
+            .epoch_length
+            .0 = 1;
+        genesis.body.consensus_state.next_epoch_data.seed =
+            EpochSeed::from_base58("2vaRh7FQ5wSzmpFReF9gcRKjv48CcJvHs25aqb3SSZiPgHQBy5Dt")
```
Comment (noot) [inline consensus/tests/checkpoint.rs:30]: is this arbitrary?

File: consensus/src/checkpoint.rs [hunk 4]
```diff
@@ -3,57 +3,18 @@
...
-    genesis.body.consensus_state.next_epoch_data.epoch_length.0 = 2;
-    Ok(())
-}
+use crate::error::ConsensusError;
 
 pub fn is_short_range(
     c0: &ProtocolStateChain,
     c1: &ProtocolStateChain,
-) -> Result<bool, ConsensusErrTyp> {
+) -> Result<bool, ConsensusError> {
```
Comment (polarbeargo) [inline consensus/src/checkpoint.rs:11]: Awesome work on standout help defines `ConsensusError` 🥇 .

File: consensus/src/common.rs (hunk 2 above)
Comment (polarbeargo) [reply consensus/src/common.rs:121]: Great work on helping modify this part 👍 !

File: consensus/src/checkpoint.rs [hunk 5]
```diff
@@ -3,57 +3,18 @@
...
-        .body
-        .consensus_state
-        .next_epoch_data
-        .start_checkpoint = StateHash::default();
-    genesis.body.consensus_state.next_epoch_data.lock_checkpoint =
-        StateHash::from_base58("3NLoKn22eMnyQ7rxh5pxB6vBA3XhSAhhrf7akdqS6HbAKD14Dh1d")
-            .map_err(|_| ConsensusErrTyp::ConsensusInitFail)?;
-    genesis.body.consensus_state.next_epoch_data.epoch_length.0 = 2;
-    Ok(())
-}
```
Comment (polarbeargo) [inline consensus/src/checkpoint.rs:45]: Uncertain if we can totally ignore this lifecycle for checkpointing?

File: consensus/src/error.rs [hunk 6]
```diff
@@ -13,6 +13,4 @@ pub enum ConsensusError {
     ConsensusStateNotFound,
```
Comment (polarbeargo) [inline consensus/src/error.rs:13]: Beautiful work on helping define Consensus in a consistent way 🚀 !

File: consensus/tests/checkpoint.rs (hunk 3 above)
Comment (creativcoder) [reply consensus/tests/checkpoint.rs:30]: No, it's from the spec: https://github.com/MinaProtocol/mina/blob/02dfc3ff0160ba3c1bbc732baa07502fe4312b04/docs/specs/consensus/README.md#531-genesis-checkpoints

File: consensus/src/checkpoint.rs (hunk 5 above)
Comment (creativcoder) [reply consensus/src/checkpoint.rs:45]: I could not follow the question, could you rephrase?

Comment (polarbeargo) [reply consensus/src/checkpoint.rs:45]: I'm wondering if in Blockchain Mina protocol checkpoint has a life cycle as [Life Cycle of a Servlet](https://www.geeksforgeeks.org/life-cycle-of-a-servlet/) has to init(), resume(),  and destroy()? Since we have init_checkpoints() to init its states. If yes we will need to handle the life cycle with care. Probably will not just throw it into the test module.

Comment (creativcoder) [reply consensus/src/checkpoint.rs:45]: No they are not related to servlet lifecycle methods. As far as i see it, the init_checkpoint function was just a helper function created to make writing test cases easier by helping stub the staking epoch and next epoch fields.

File: consensus/src/checkpoint.rs (hunk 1 above)
Comment (hanabi1224) [reply consensus/src/checkpoint.rs]: Sounds good, thx!

**Review by polarbeargo — APPROVED** [review]
Thank you for helping refactors, Excellent work 🚀
